### PR TITLE
Feat: Modularize the code and remove dangling channels and outputs

### PR DESCRIPTION
**Pull request by abhi18av**
Hi @thorellk and @emilio-r 

Continuing from our conversation on https://github.com/nextflow-io/nextflow/issues/1796, I've tried to modernize the code as per best practices of DSL2. I hope you find them as a cleaner way to organize code.

The contents for the `BACTpipe_results` folder is the same when I tested, but I'd appreciate if you could also try out this branch and make sure things work as expected.

The DAG for `3.E` branch 

image

The DAG for my `feat/dsl2` branch

image

Beyond the current changes, here's a list of improvement I can think of

[ ] Localize the `bin` directory to a module once https://github.com/nextflow-io/nextflow/issues/1798 is resolved
[ ] Update the docs 
[ ] Add a container oriented profile
[ ] Make the pipeline [Tower](tower.nf/) friendly
[ ] Optionally, add unit and integration tests, you can refer an example [here](https://github.com/seqeralabs/gatk4-germline-snps-indels/)

I have a request, would it be possible to add me as a maintainer of this pipeline? I'd to be happy to keep `BACTpipe` updated as I have a [personal interest in Bacterial genomics](https://www.researchgate.net/profile/Abhinav_Sharma79) :)

### CODE REVIEW

**Comment by abhi18av** [comment]
Please review the code changes and once you feel these changes are acceptable, we can go ahead.

File: modules/prokka/prokka.nf
Comment (boulund) [inline modules/prokka/prokka.nf:27]:
```suggestion
        --compliant \
```

**Comment by boulund** [comment]
Awesome that you wanted to spend the time! Much appreciated, BACTpipe was sorely in need of some love to bring it up-to-speed with recent Nextflow developments. I'll let @thorellk and @emilio-r review the PR and test the branch before we decide to merge. 

Very kind of you to offer taking on a maintainer role for the project as well! I think it could be a great idea, we just need to spend some time to discuss what our ambitions for the project are and how we would like to see it develop, to make sure our ideas align :)

**Comment by abhi18av** [comment]
>  I think it could be a great idea, we just need to spend some time to discuss what our ambitions for the project are and how we would like to see it develop, to make sure our ideas align :)

@boulund, that's cool with me :)

**Review by thorellk — APPROVED** [review]
I think this looks reasonable from my limited knowledge of nextflow. Thank you @abhi18av

**Comment by abhi18av** [comment]
@thorellk , you're welcome.

> Very kind of you to offer taking on a maintainer role for the project as well! I think it could be a great idea, we just need to spend some time to discuss what our ambitions for the project are and how we would like to see it develop, to make sure our ideas align :)

Once you've had a word with the team, please let me know of your thoughts here.

**Comment by abhi18av** [comment]
Thanks for the invite @boulund 😊

**Comment by boulund** [comment]
Welcome! Glad to have you here to join us! :)

**Pull request merged**
